### PR TITLE
[SLM] iOS refactor

### PR DESCRIPTION
**Pull request by davidpissarra**
This PR refactors the iOS app to fully support SLM, specifically addressing the following items:

- [X] Fetch `modelID` and `modelLib` from `app-config.json` instead of `mlc-chat-config.json`
- [X] Refactor `app-config.json`
- [X] Update `ModelRecord` schema accordingly
- [X] Update `localID` to `modelID`
- [x]  Update docs

### CODE REVIEW

**Comment by davidpissarra** [comment]
PR now ready for review. cc @CharlieFRuan @tqchen

File: docs/prebuilt_models.rst [hunk 1]
```diff
@@ -466,7 +466,7 @@ Mistral
     -
```
Comment (CharlieFRuan) [inline docs/prebuilt_models.rst:466]: Could we make corresponding changes and uncomment 
https://github.com/mlc-ai/mlc-llm/blob/main/docs/prebuilt_models.rst?plain=1#L98-L210

The idea is to give users some quick steps to follow for using prebuilt models on iOS and point to the main iOS page, something similar to Android. We can refer to the previous prebuilt_model page: https://llm.mlc.ai/docs/prebuilt_models_deprecated.html#prebuilt-models-on-ios.

File: ios/MLCChat/app-config.json [hunk 2]
```diff
@@ -1,27 +1,18 @@
...
-  ],
   "model_list": [
     {
-      "model_url": "https://huggingface.co/mlc-ai/mlc-chat-Llama-2-7b-chat-hf-q3f16_1/",
-      "local_id": "Llama-2-7b-chat-hf-q3f16_1"
+      "model_path": "Mistral-7B-Instruct-v0.2-q3f16_1",
+      "model_id": "Mistral-7B-Instruct-v0.2-q3f16_1",
+      "model_lib": "mistral_q3f16_1",
+      "model_lib_path": "lib/Mistral-7B-Instruct-v0.2/Mistral-7B-Instruct-v0.2-q3f16_1-iphone.tar",
+      "estimated_vram_bytes": 4000000000
```
Comment (CharlieFRuan) [inline ios/MLCChat/app-config.json:8]: I wonder if we could have a more specific number using https://github.com/mlc-ai/mlc-llm/pull/1525. The estimate is calculated based on the metadata, so `-cuda.so` would give the same estimation.

File: docs/deploy/ios.rst [hunk 3]
```diff
@@ -148,18 +148,28 @@ Customize the App
 
 We can customize the iOS app in several ways.
```
Comment (CharlieFRuan) [inline docs/deploy/ios.rst:149]: Could we add an end-to-end tutorial for:
1. Adding a model variant (e.g. adding OpenHermes -- hence no need to compile model)
2. Adding a new model (e.g. adding RedPajama -- hence need to compile model)

We could just give some concise steps, then ask user to refer to our [convert_weight](https://llm.mlc.ai/docs/compilation/convert_weights.html) and [compile_model](https://llm.mlc.ai/docs/compilation/compile_models.html) pages correspondingly. 

For web-llm, these correspond to:
1. https://llm.mlc.ai/docs/deploy/javascript.html#bring-your-own-model-variant
2. https://llm.mlc.ai/docs/deploy/javascript.html#bring-your-own-model-library

**Review by CharlieFRuan — COMMENTED** [review]
Thank you so much for the work! Left some nit, primarily on the documentation.

File: ios/MLCChat/app-config.json (hunk 2 above)
Comment (davidpissarra) [reply ios/MLCChat/app-config.json:8]: That makes sense, I will run that command to better report estimated mem usage. Thanks!

File: docs/deploy/ios.rst (hunk 3 above)
Comment (davidpissarra) [reply docs/deploy/ios.rst:149]: Sounds great, I will work on that in the meantime

File: docs/deploy/ios.rst [hunk 4]
```diff
@@ -176,6 +176,257 @@ This built-in list can be controlled by editing ``prepare_params.sh``.
...
+.. code:: javascript
+   
+   "model_list": [
+      // Other records here omitted...
+      {
+         // Substitute model_url with the one you created `my-huggingface-account/my-mistral-weight-huggingface-repo`
+         "model_url": "https://huggingface.co/mlc-ai/NeuralHermes-2.5-Mistral-7B-q3f16_1-MLC",
+         "model_id": "Mistral-7B-Instruct-v0.2-q3f16_1",
+         "model_lib": "mistral_q3f16_1",
+         "model_lib_path": "lib/Mistral-7B-Instruct-v0.2/Mistral-7B-Instruct-v0.2-q3f16_1-iphone.tar",
```
Comment (tqchen) [inline docs/deploy/ios.rst:273]: is model_lib_path necessary? I feel this is snot something we need during runtime

File: docs/deploy/ios.rst [hunk 5]
```diff
@@ -148,24 +148,287 @@ Customize the App
...
+``model_lib``
+   (Required) Matches the system-lib-prefix, generally set during ``mlc_chat compile`` which can be specified using 
+   ``--system-lib-prefix`` argument. By default, it is set to ``"${model_type}_${quantization}"`` e.g. ``gpt_neox_q4f16_1`` 
+   for the RedPajama-INCITE-Chat-3B-v1 model. If the ``--system-lib-prefix`` argument is manually specified during 
+   ``mlc_chat compile``, the ``model_lib`` field should be updated accordingly.
+
+``model_lib_path``
+   (Required) Path to the model library (``.tar`` file) in the repo. Only used while running ``prepare_libs.sh`` to determine
+   which model library to use during runtime. Useful when selecting a library with different settings (e.g. ``prefill_chunk_size``,
+   ``context_window_size``, and ``sliding_window_size``).
```
Comment (davidpissarra) [inline docs/deploy/ios.rst:171]: made `model_lib_path` doc more explicit, but we can also change the naming to make it more intuitive

cc @tqchen

Comment (tqchen) [reply docs/deploy/ios.rst:171]: let us keep this name for now, let us clarify what is the root of the path(e.g. how the system searchs for these libs), cc @Kartik14 @CharlieFRuan to update android side docs as well

Comment (davidpissarra) [reply docs/deploy/ios.rst:172]: Should be better clarified now

**Review by CharlieFRuan — APPROVED** [review]
LGTM, thank you!

**Comment by tqchen** [comment]
Let us followup with the convention update per https://github.com/mlc-ai/mlc-llm/pull/1659

**Pull request merged**